docs/adrs/0274-step-conclusion.md

```diff
@@ -0,0 +1,43 @@
+# ADR 0274: Step conclusion
+
+**Date**: 2020-01-13
+
+**Status**: Accepted
+
+## Context
```

Comment:
@chrispat do you have any good scenarios offhand to add to the ADR?

Reply:
The main scenario is running another step if a specific step passes or fails.

Reply:
do you have a good example? i'm looking to add a good example in the adr

Reply:
How about a step that generates an artifact if if succeedes but doesn't if it fails and then you want to condition the upload artifact on that conclusion.

Reply:
@chrispat wouldnt the job be in a failed state at that point? so downstream upload artifact step wouldnt run anyway

Reply:
what about?
```
steps:
- id: genpkg
  uses: myorg/genpkg@v1
- id: testpkg
  uses: myorg/testpkg@v1
- id: uploadpkg
  if: steps.genpkg.conclusion == success
  uses: actions/upload-artifact@v1
```

Reply:
@TingluoHuang Why even bother with the condition then? Without a condition, if `genpkg` fails then downstream steps without a condition will not execute.

In order to meet the scenarios, due to continue-on-error flipping failed to success, I think exposing issues is the key.

So this scenario would work:

```yaml
- id: experimental-proj
  continue-on-error: true
  run: ./build.sh experimental-proj
- if: ${{ !steps.experimental-proj.issues[0] }}
  run: ./publish.sh experimental-proj
```

Reply:
What if I want to explicitly run a step if a previous step was skipped?

Reply:
Good point. Sorry I misspoke. I think exposing conclusion _and_ issues is the key to covering these types of scenarios.

Reply:
Although technically i guess you could use the same condition the previous step did, if you also want to skip :P

Reply:
`skipped` only caused by `if` return `false`, in that case you can just add `!` to the condition that skipped the step?

```
- run: printenv
  if: whatever condition
- run: ls -l
  if: !(whatever condition)
```

Reply:
Pushed updates to the ADR with `outcome` since the continue-on-error value is more useful.

Ting and I played with what adding issues to the context would look like. It adds a lot of complexity for users to deal with.

We feel `outcome` (pre continue-on-error value) solves the most common scenarios.

Reply:
@chrispat for thoughts about this latest proposal ^